app/controllers/application_controller.rb

```diff
@@ -5,6 +5,130 @@ class ApplicationController < Sinatra::Base
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "password_security"
   end
 
-end
\ No newline at end of file
+  get '/' do
+    erb :index
+  end
+
+  get '/tweets' do
+    @user = current_user if logged_in?
+    logged_in? ? (erb :'tweets/tweets') : (redirect to '/login')
+  end
+
+  get '/tweets/new' do
+    logged_in? ? (erb :'tweets/create_tweet') : (redirect to '/login')
+  end
+
+  post '/tweets' do
+    if logged_in? && (current_user.id == session[:id])
+      if params[:content].strip.empty?
+        redirect to '/tweets/new'
+      else
+        params[:user_id] = current_user.id
+        @tweet = Tweet.create(params)
+        redirect to '/tweets'
+      end
+    else
+      redirect to '/'
+    end
+  end
+
+  get '/tweets/:id' do
+    if logged_in?
+      @tweet = Tweet.find_by_id(params[:id])
+      !!@tweet ? (erb :'tweets/show_tweet') : (redirect to '/tweets')
+    else
+      redirect to '/login'
+    end
+  end
+
+  get '/tweets/:id/edit' do
+    @tweet = Tweet.find_by_id(params[:id])
+    if logged_in?
+      if current_user.id == @tweet.user_id
+        !!@tweet ? (erb :'tweets/edit_tweet') : (redirect to '/tweets')
+      else
+        redirect to '/tweets'
+      end
+    else
+      redirect to '/login'
+    end
+  end
+
+  patch '/tweets/:id' do
+    if params[:content].strip.empty?
+      redirect to "/tweets/#{params[:id]}/edit"
+    else
+      if !!Tweet.find_by_id(params[:id])
+        @tweet = Tweet.find_by_id(params[:id])
+        @tweet.content = params[:content]
+        @tweet.save
+      end
+      redirect to "/tweets/#{@tweet.id}"
+    end
+  end
+
+  delete '/tweets/:id/delete' do
+    @tweet = Tweet.find_by_id(params[:id])
```

Comment:
Prefer to find tweets that are already only associated to the `current_user` than to manually check the `user_id` foreign key of the tweet against the `current_user.id`

```ruby
tweet = current_user.tweets.find_by(id: params[:id])
if tweet && tweet.destroy
  redirect "/tweets"
else
  redirect "/tweets/#{tweet.id}"
end
```

We're using the association collection method provided by [`has_many`](http://api.rubyonrails.org/classes/ActiveRecord/Associations/ClassMethods.html#method-i-has_many) to query the user's tweets. 

Note the usage of [`find_by`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find-by) vs [`find`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find). Had we used `find` here, ActiveRecord would raise an error in the event that we could not find a tweet associated with this user. `find_by` will allow this action to fail gracefully, simply returning nil if the user is trying to delete a tweet that they do not have access to.